src/test/java/CPPParserTest.java

```diff
@@ -0,0 +1,103 @@
+import org.junit.Assert;
+import org.junit.Test;
+import org.prop4j.And;
+import org.prop4j.Literal;
+import org.prop4j.Node;
+import org.variantsync.diffdetective.datasets.DatasetDescription;
+import org.variantsync.diffdetective.datasets.DatasetFactory;
+import org.variantsync.diffdetective.diff.GitPatch;
+import org.variantsync.diffdetective.diff.difftree.DiffTree;
+import org.variantsync.diffdetective.diff.difftree.parse.IllFormedAnnotationException;
+import org.variantsync.diffdetective.diff.result.DiffError;
+import org.variantsync.diffdetective.feature.CPPAnnotationParser;
+import org.variantsync.diffdetective.feature.CPPDiffLineFormulaExtractor;
+import org.variantsync.functjonal.Result;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class CPPParserTest {
+    private record TestCase(
+            String diffLine,
+            String expectedExtractionResult,
+            Node expectedFormula
+    ) {}
+
+    private static final Path reposPathWSL = Paths.get("\\\\wsl$", "Ubuntu/home/bittner/VariantSync", "DiffDetectiveMining");
```

Comment:
This might also make your convenience constructors obsolete.

Reply:
Oops. The convencience constructors are a handy tool for future debugging, so I would still keep them. I would have needed them a lot in the past.

I agree to the proposed change but we can just remove this test class anyway because you provided an extensive test suite for the diff parser. So this test case here is not needed anymore?

Reply:
I kept the convenience methods, but replaced all the tests with simple examples (no reference to some outside source code).